### PR TITLE
Changes made during merge

### PR DESCRIPTION
**Pull request by jpaddison3**
The base branch is a branch that was the result of merging the `master` branch (which has a version of LW master from two weeks ago) into `ea-forum-devel`, which is ~the code running on production. I intend to keep this pattern.

I expect most things are pretty self-explanatory. One selected change is that hasEvents was previously a database setting, but needs to be an instance setting if I'm going to be switching back and forth between running my localhost as a LW site and EA site for testing.

### CODE REVIEW

**Comment by colophonemes** [comment]
I've run the `loadDevDbSettings.js` script (there was no output other than noting the connection string, but I assume this means it ran OK). Getting the following console error on startup, 

```
W20200730-11:55:54.924(1)? (STDERR) Database ID in config file and in database don't match.
W20200730-11:55:54.924(1)? (STDERR)     Database ID: undefined
W20200730-11:55:54.924(1)? (STDERR)     Expected database ID: dragonair-dev
W20200730-11:55:54.924(1)? (STDERR) If you are connecting to a production DB, you must use a matching config file. If you are *not* connecting to a production DB, you should not use a production config file.
=> Exited with code: 1
=> Your application is crashing. Waiting for file change.
```

File: packages/lesswrong/components/Layout.tsx [hunk 1]
```diff
@@ -59,7 +59,7 @@ const styles = theme => ({
     paddingBottom: 15,
     marginLeft: "auto",
     marginRight: "auto",
-    background: "#f4f4f4",
+    background: "#f6f8f9",
```
[the remark this answers is not in the document]
Comment (colophonemes) [reply packages/lesswrong/components/Layout.tsx:62]: Feels like this should be `theme.palette.background.default` for DRYness

[the remark this answers is not in the document]
Comment (jpaddison3) [reply packages/lesswrong/components/Layout.tsx:62]: Indeed, LW makes that update soon. I didn't depart from their mistake here.

**Comment by colophonemes** [comment]
Cool, that works, but now leads to the following console output, which might be fine:
```
No setting value provided for public instance setting for setting with name expectedDatabaseId despite it being marked as warning
```

**Comment by colophonemes** [comment]
Trying to subscribe via email, but all feeds except `Curated` are greyed out:

Screen Shot 2020-07-30 at 12 10 22 pm

File: packages/lesswrong/components/Layout.tsx [hunk 2]
```diff
@@ -273,7 +273,7 @@ class Layout extends PureComponent<LayoutProps,LayoutState> {
         .includes(currentRoute?.name)
         
     const shouldUseGridLayout = standaloneNavigation
-
+
```
[the remark this answers is not in the document]
Comment (jpaddison3) [reply packages/lesswrong/components/Layout.tsx:276]: No, LW style prefers that empty lines between indented lines should have whitespace up to the indent level. I think it's dumb, but to maintain consistency, I spent an unreasonable amount of time configuring my editor to follow that style and hopefully avoid whitespace-only diffs. Unfortunately they did not follow that style here, so when I added and removed a console log, I introduced a change.

**Comment by jpaddison3** [comment]
Curated shouldn't even be there. Thanks.

File: settings-prod.json [hunk 3]
```diff
@@ -1,86 +1,20 @@
...
-TR3iCZk+fqGEacWnKli09dUrnFx4LL1QbrIAufs+CeFA+ACHUyGrFppbX3+qPxFs
-OUDFsKuKo3owUDeH3TPXz4BpbAWjaGGTjkckHn/zB+gHIWm7OVVao8X9/rChmM0s
-w/qFPx6QcnVwANiq6Ukfu04q/DZxUz5X+bf3hJH0wkAKvOjdUFqYTU5/AY1pft1F
-Ccbjrw5M2PiYrks/rncu9dcZtPux/lG0ousB3/r/2RLoFqEedrqpu0M20lVTo0gr
-x48omU8cWTWyEaJuDwzE+XtnT8QTP2kPzVpCYfRixbZK19O7VaPh+9EnZ8mJIOml
-QbAejeVO3JOI/McfF6gTeWMqYUcaX66b4X984gDKC4YoRj9YjwaxPv41PLNL+8eU
-FhplfjhRpGujQibjMh93qRQDR//ZjmdeFq5hYCjXbe7+8udh8BPaHEh7xllPcesK
-u9shylCjbK+iieZLylMUBNxdLUUtwsoA4gafDd7gt8XUhN0il4r3gJHoFuG5rSBO
-1x5+iD2UatU3z2Op0Zsf/A==
+U2FsdGVkX18C+Bd+jgS1q5pP0UghspxDk6YmiYK8kDtxd7LS6s0Hg2xCjQuuEvE+
```
[the remark this answers is not in the document]
Comment (jpaddison3) [reply settings-prod.json:1]: That shouldn't do it.

**Comment by colophonemes** [comment]
If I try to use the sunshine sidebar to tag something, the hover-over when I go to select a tag is just an infinite loader:

image

**Comment by colophonemes** [comment]
If I try to create a new tag, I get the following error:

image


image

**Comment by jpaddison3** [comment]
Unfortunately, search on your dev machine is going to be a bad time. Algolia is like, "what, I got a webhook telling me to add kitten pictures to my tag index, here's the tag id". And your local database is like "I don't got one".

**Comment by jpaddison3** [comment]
Ah, you should run migrations. Please hold

**Comment by colophonemes** [comment]
Maybe there's a migration I should run?

**Comment by jpaddison3** [comment]
Jynx. Here they are:
```
> Vulcan.migrations.deletedNotifications()
> Vulcan.migrations.clearOldPartiallyReadSequences()
> Vulcan.migrations.populateTagUser()
> Vulcan.migrations.setDirectChildrenCount()
```
As a reminder, run those from `meteor shell`.

**Comment by jpaddison3** [comment]
You might need to run more migrations if it's been a long time.
image

**Comment by colophonemes** [comment]
Ran the migrations, but still get the same error

**Comment by colophonemes** [comment]
>You might need to run more migrations if it's been a long time.

How do I see that view?

**Comment by jpaddison3** [comment]
/admin/migrations

**Comment by colophonemes** [comment]
Haha looks like it:

image

**Comment by colophonemes** [comment]
OK, have run all the migrations, and restarted my server, but still getting the same error when trying to save tags. I'm willing to believe this is a borked DB on my machine, but would be good if you can try to replicate on your local install.

**Comment by jpaddison3** [comment]
Found it. The new tag thing should not be visible to users who are not opted into beta.

**Comment by jpaddison3** [comment]
And fixed

File: packages/lesswrong/components/common/RecaptchaWarning.tsx [hunk 4]
```diff
@@ -26,12 +26,21 @@ const RecaptchaWarning = ({ currentUser, classes, children }: {
...
     return <> { children } </>
   }
-  // ea-forum look here: You will want to change where this links to
-  return <div className={classes.warningText}>
-    You've been flagged by our spam detection system. Please message an admin via 
-    Intercom (the chat bubble in the bottom right corner) or send a private message to admin 
-    <Link className={classes.link} to="/users/habryka"> habryka</Link> to activate posting- and commenting-privileges on your account.
-  </div>
+  switch (forumTypeSetting.get()) {
+    case 'EAForum':
```
[the remark this answers is not in the document]
Comment (jpaddison3) [reply packages/lesswrong/components/common/RecaptchaWarning.tsx:30]: I think there's a way to force recaptcha to fail. I tested it by commenting out lines 26-28 above.

File: packages/lesswrong/components/ea-forum/EAHomeHandbook.tsx [hunk 5]
```diff
@@ -17,7 +17,7 @@ const bannerHeight = 250
 const styles = createStyles(theme => ({
   bannerContainer: {
     position: 'absolute',
-    top: 120, // desktop header height + layout margin
+    top: 130, // desktop header height + layout margin + negative margin
```
[the remark this answers is not in the document]
Comment (jpaddison3) [reply packages/lesswrong/components/ea-forum/EAHomeHandbook.tsx:20]: If you login as an admin and clear cookies.

**Pull request merged**